File: src/python/calculate_pi.py
""" Calculating PI in Python """


def calculate_pi(number):
    """
    Implementation of a PI calculation algorithm.

    Arguments:
    number: int.

    Returns the value of PI.
    """
    denominator = 1.0
    operation = 1.0
    pi = 0.0
    for _ in range(number):
        pi += operation * (4.0 / denominator)
        denominator += 2.0
        operation *= -1.0
    return pi


if __name__ == "__main__":
    n_terms = [10, 1000, 100000, 10000000]
    for n in n_terms:
        print(f"PI ({n}): {calculate_pi(n)}")

File: src/swift/calculatePi.swift
//
//  calculatePi.swift
//  
//
//  Created by Matheus Torres on 14/11/20.
//

func calculatePi(_ number: Int) -> Float {
    var denominator: Float = 1
    var operation: Float = 1
    var pi: Float = 0
    
    for _ in 0..<number {
        pi += operation * (4 / denominator)
        denominator += 2
        operation *= -1
    }
    
    return pi
}

print("PI 10: \(calculatePi(10))")
print("PI 1000: \(calculatePi(1000))")
print("PI 100000: \(calculatePi(100000))")
print("PI 10000000: \(calculatePi(10000000))")

